dspace-xmlui/src/main/java/org/dspace/app/xmlui/cocoon/SearchMetadataExportReader.java

```diff
@@ -0,0 +1,367 @@
+                    HttpServletResponse httpResponse = (HttpServletResponse)objectModel.get(HttpEnvironment.HTTP_RESPONSE_OBJECT);
+                    httpResponse.sendRedirect(redictURL);
+                    return;
+                }
+            }
+            else if(search_export_config.equals("anonymous")) {
+            	csv = exportMetadata(context, objectModel, query, scope, filters);
+                filename = "search-results.csv";
+            }
+        }
+        catch (RuntimeException e) {
+            throw e;    
+        }
+        catch (IOException e) {
+        	throw new ProcessingException("Unable to export metadata.",e);
+        }
+        catch (Exception e) {
+            throw new ProcessingException("Unable to read bitstream.",e);
+        } 
+    }
+    
+    /**
+	 * Write the CSV.
+	 */
+    public void generate() throws IOException, SAXException, ProcessingException
+    {
+        response.setContentType("text/csv; charset=UTF-8");
+        response.setHeader("Content-Disposition","attachment; filename=" + filename);
+        out.write(csv.toString().getBytes("UTF-8"));
+        out.flush();
+        out.close();
+    }
+    
+    /**
+	 * Recycle
+	 */
+    public void recycle() {        
+        this.response = null;
+        this.request = null;
+    }
+    
+    /**
+     * Save and return the search results as a csv file
+     * 
+     * @params context, objectModel, query, scopeString, filters
+     * 
+     * @throws IOException, UIException, SearchServiceException, SQLException
+     */
+    public DSpaceCSV exportMetadata(Context context, Map objectModel, String query, String scopeString, String filters) throws IOException, UIException, SearchServiceException, SQLException
+    {
+    	DiscoverResult qResults = new DiscoverResult();
+    	DiscoverQuery qArgs = new DiscoverQuery();
+        
+    	try {
+    		scopeString = scopeString.replace("~", "/");
+        }
+        catch(NullPointerException e) { }
+    	
+        // Are we in a community or collection?
+        DSpaceObject scope;
+        if (scopeString == null || "".equals(scopeString)) {
+            // get the search scope from the url handle
+        	scope = HandleUtil.obtainHandle(objectModel);
+        }
+        else {
+            // Get the search scope from the location parameter
+        	scope = HandleManager.resolveToObject(context, scopeString);
+        }
+        
+    	List<String> filterQueries = new ArrayList<String>();
+
+        String[] fqs = filters.split(",");
+        
+        if (fqs != null) {
+            filterQueries.addAll(Arrays.asList(fqs));   
+        }
+        
+        // some arbitrary value for first search
+        qArgs.setMaxResults(10);
+
+        //Add the configured default filter queries
+        DiscoveryConfiguration discoveryConfiguration = SearchUtils.getDiscoveryConfiguration(scope);
```

Comment:
This code is copied from AbstractSearch. It is very likely that this won't give exactly the same results as you are seeing in the UI. If this gives the same results now, someone will change something in the future in abstract search. This probably means that abstractsearch needs refactoring to extract the code so you can just call it here and have a guarantee that the search is identical.